### PR TITLE
PR: Update plugin to work with Spyder 4

### LINKED ISSUE #43: Vim in Spyder 4
I wanted to know if this plugin is compatible with spyder 4.

I installed it in spyder 3 and it works correctly, but when I install it in spyder 4 the vim bar does not appear to me.

### PR DESCRIPTION
**Pull request by bgallois**
Updated plugin to work with Spyder 4, fixes #43

### CODE REVIEW

**Comment by goanpeca** [comment]
@bgallois thanks for this!

Will review this sometime this week. Thanks again :-)

File: spyder_vim/tests/test_vim.py [hunk 1]
```diff
@@ -24,7 +24,12 @@
 
 # Spyder imports
 # from spyder.utils.fixtures import setup_editor
-from spyder.widgets.editor import EditorStack
+try:
+    # Spyder 4
+    from spyder.plugins.editor.widgets.editor import EditorStack
+except ImportError:
+    # Spyder 3
+    from spyder.widgets.editor import EditorStack
```
Comment (goanpeca) [inline spyder_vim/tests/test_vim.py:32]: @ccordoba12 should we remove spy3.x compatibility ?

**Comment by goanpeca** [comment]
I think we will probably remove Spyder 3 compat, besides that looks good!

File: spyder_vim/tests/test_vim.py [hunk 2]
```diff
@@ -24,7 +24,12 @@
 
 # Spyder imports
 # from spyder.utils.fixtures import setup_editor
```
Comment (goanpeca) [inline spyder_vim/tests/test_vim.py:26]: we can remove this line if we are not using it

File: spyder_vim/tests/test_vim.py [hunk 3]
```diff
@@ -34,6 +39,11 @@
 LOCATION = osp.realpath(osp.join(
     os.getcwd(), osp.dirname(__file__)))
 
+class VimTesting(Vim):
```
Comment (goanpeca) [inline spyder_vim/tests/test_vim.py:42]: Please add a space above here

File: spyder_vim/tests/test_vim.py [hunk 4]
```diff
@@ -34,6 +39,11 @@
...
+    CONF_FILE = False
+
+    def __init(self, parent):
+        Vim.__init__(self, parent)
```
Comment (goanpeca) [inline spyder_vim/tests/test_vim.py:46]: Please add a space below here

File: spyder_vim/vim.py [hunk 5]
```diff
@@ -42,10 +50,11 @@ def get_plugin_icon(self):
 
     def register_plugin(self):
         """Register plugin in Spyder's main window."""
+        super(Vim, self).register_plugin()
         self.focus_changed.connect(self.main.plugin_focus_changed)
         self.vim_cmd.editor_widget.layout().addWidget(self.vim_cmd)
-        fixed_shortcut("Esc", self.vim_cmd.editor_widget.editorsplitter,
-                       self.vim_cmd.commandline.setFocus)
+        sc = QShortcut("Esc", self.vim_cmd.editor_widget.editorsplitter, self.vim_cmd.commandline.setFocus)
```
Comment (goanpeca) [inline spyder_vim/vim.py:56]: Is this "Esc" a translatable string @bgallois ?

**Comment by bgallois** [comment]
@goanpeca thanks! I fixed the formatting and the key sequence in the shortcut.
The spyder 3 compatibility is removed from the test and fixed for the plugin.

**Pull request merged**